### PR TITLE
Changes for accomodating staging and daily environments

### PR DESCRIPTION
**Pull request by amarathavale**
We want to make this change to handle the following scenarios
1. Create Collection if one doesn't exist. In staging, a collection will always be present, and we want to only apply the desired configuration to the existing collection
2. Loading data only if we don't have the intended number of documents. In daily, when a collection is created we would have to load N documents. But in staging, since the collection isn't deleted/recreated on each run, we only want to load the incremental documents.

Changed the benchmark script to allow configuring the JVM heap size.

Validated the changes using the java cli, and also the docker image run locally.

### CODE REVIEW

**Review by moderakh — APPROVED** [review]
please wait for @simplynaveen20 review and sign off.

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CollectionResourceManager.java
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CollectionResourceManager.java:63]: Please check for status code 200 as well

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CollectionResourceManager.java
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CollectionResourceManager.java:86]: We should check status code 404 on exception to determine collection does not exist, and for any other exception we should float the exception up

Comment (amarathavale) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CollectionResourceManager.java:86]: I had started down this path, but there are 2 possibilities in the event of an error (and in both cases it returns Optional.empty()
1. Collection does not exist
2. Collection exists, but there was another error

In either case, the code will ahead and call createContainer, which uses createContainerIfNotExists to create the Collection. If there is an error on this call, then we throw the exception all the way up.

To put my long answer differently: What's the impact/consequence of not checking for 404?

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CollectionResourceManager.java
Comment (amarathavale) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CollectionResourceManager.java:63]: The presence/not present is handled with getContainerProperties. Any checks will have to happen there. Resolving this convo and moving to the other open conversation.

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CollectionResourceManager.java
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CollectionResourceManager.java:86]: "Collection exists, but there was another error" in this case error could be anything so it is ideal to float the error right there instead of going further in the logic. In this case it is not a big deal as this is testing workload, so your call. But it is advisable to do the right error check in real prod application.

**Review by simplynaveen20 — APPROVED** [review]
Approved with the advisable comment on non found scenario.

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CollectionResourceManager.java
Comment (amarathavale) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CollectionResourceManager.java:86]: I agree if there is an error, there's not much value in continuing.

**Pull request merged**
